src/Domain/CampaignRepository.php

```diff
@@ -110,10 +117,16 @@ private function pullCharity(
 
         $tbgCanClaimGiftAid = (
             !empty($hmrcReferenceNumber) &&
-            in_array($giftAidOnboardingStatus, static::$giftAidOnboardedStatuses, true)
+            in_array($giftAidOnboardingStatus, self::GIFT_AID_ONBOARDED_STATUSES, true)
+        );
+        $tbgApprovedToClaimGiftAid = (
+            !empty($hmrcReferenceNumber) &&
+            in_array($giftAidOnboardingStatus, self::GIFT_AID_APPROVED_STATUSES, true)
         );
 
         $charity->setTbgClaimingGiftAid($tbgCanClaimGiftAid);
+        $charity->setTbgApprovedToClaimGiftAid($tbgApprovedToClaimGiftAid);
```

Comment:
Maybe at this point it's worth moving this code into the Charity class - something like https://github.com/thebiggive/matchbot/commit/11db11a63b36f6a8637440ab81874a5a7f876b0a . That would allow making several of the setters private or removing them.

Or would it make sense now that we pull quite a lot from SF to convert CharityRepository into a SalesforceReadyProxyRepository? Maybe the reasoning at https://github.com/thebiggive/matchbot/blob/1dd8998c64f0c81d4a25c80694cb1062a579edb9/src/Domain/CharityRepository.php#L10 is out of date.

Reply:
It's true the comment is out of date in terms of its list of fields, but I think the more important rationale – reducing complexity and load in Salesforce – probably still stands. In particular we don't want to double how many distinct calls to Salesforce it takes to keep MatchBot up to date because that's a high risk area in terms of sort limits that are hard to usefully load test against.

I'll amend the comment but I don't think we'd want to make Charity a totally independent proxy with its own Salesforce endpoint in the near future.

Reply:
@NoelLH That makes sense, updating the comment would be useful.

Reply:
Did the move - may be still worth unit testing though? not sure.